src/components/screens/Home.tsx

```diff
@@ -2,58 +2,64 @@ import { useEffect, useState } from 'react';
 import PageContainer from '../containers/PageContainer';
 import AddStocksForm from '../forms/AddStocksForm';
 import AccountStats from '../shared/AccountStats';
-import InvestmentCard from '../shared/InvestmentCard';
 import { Stock } from '~/clients/firebase-client/models/Investments';
 import { joinStockData } from '~/helpers/join-stock-data';
 import { Dialog } from '@headlessui/react';
-import { invetuClient } from '~/clients/invetu-client/invetu-client';
 import { Head } from '../shared/Head';
 import RadialChart from '../shared/RadialChart';
-import { HistoryAPI } from '~/clients/invetu-client/models/HistoryAPI';
-import Dividends from '../shared/Dividends';
 import EvolutionChart from '../shared/EvolutionChart';
 import { useCustomSelector } from '~/hooks/use-custom-selector';
 import { useDispatch } from 'react-redux';
 import { fetchInvestments } from '~/features/investments/investments-slice';
 import Ghost from '~/assets/illustrations/ghost.svg';
 import Add from '~/assets/illustrations/add.svg';
 import WrapperIcon from '../shared/WrapperIcon';
-
+import { Result } from '~/clients/firebase-client/models/history-stock-br';
+import { fetchAllInvestmentsData } from '~/features/investments-data/investments-data-slice';
+import InvestmentCard from '../shared/InvestmentCard';
+import Dividends from '../shared/Dividends';
 export default function Home() {
   const [investmentsJoined, setInvestmentsJoined] = useState<Array<Stock>>([]);
   const [isOpen, setIsOpen] = useState(false);
   const [currentBalance, setCurrentBalance] = useState(0);
-  const [stocksHistory, setStocksHistory] = useState<Array<HistoryAPI>>();
+  const [stocksHistory, setStocksHistory] = useState<Array<Result>>();
   const investmentsStore = useCustomSelector(state => state.investments);
+  const investmentsDataStore = useCustomSelector(
+    state => state.investmentsData,
+  );
   const dispatch = useDispatch();
 
   useEffect(() => {
     dispatch(fetchInvestments());
   }, []);
 
+  useEffect(() => {
+    dispatch(fetchAllInvestmentsData());
+  }, []);
+
```

Comment:
Idk, review it

Reply:
Moved to the Main.tsx, don't think that it's too much better, but probably the whole application will be refactored soon, so I can handle that